app/src/main/java/supermarket/model/Discount.kt

```diff
@@ -1,3 +1,3 @@
 package supermarket.model
 
-class Discount(val product: Product, val description: String, val discountAmount: Double)
```

Comment:
Why did we remove product?  I kinda liked having it there for some of the tests I was writing.

Reply:
It's like the `SingleProductOffer` - not all Discounts are associated with one Product, so instead we now let the Discount keep track of that itself and produce its description.  Happy to discuss whatever test scenarios and how we can make them smoother!